frontend/src/pages/PostsPage.tsx

```diff
@@ -1,9 +1,76 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { useState } from "react";
+import PostComponent from "../components/PostComponent";
 
+interface Post {
+  id: number;
+  title: string;
+  content: string;
+}
 
 const PostsPage = () => {
+  const queryClient = useQueryClient();
+
+  const [title, setTitle] = useState<string>("")
+  const [content, setContent] = useState<string>("")
+
+  const {
+    data: posts,
+    isLoading,
+    isError,
+  } = useQuery<Post[]>({
+    queryKey: ["posts"],
+    queryFn: async () => {
+      const response = await fetch("http://localhost:4000/posts");
+      const data = await response.json();
+      return data;
+    },
+  });
+
+  const createPostMutation = useMutation({
+		mutationFn: async (newPost: Post) => {
+			const response = await fetch("http://localhost:4000/posts", {
+				method: "POST",
+				headers: { "Content-Type": "application/json" },
+				body: JSON.stringify(newPost),
+			});
+			const data = await response.json();
+			console.log(data);
+		},
+		onMutate: () => {},
+		onSettled: () => {},
+    onError: () => {},
+		onSuccess: () => {
+			queryClient.invalidateQueries({
+				queryKey: ["posts"],
+			});
+		},
+	});
+
+  const handleCreatePost = () => {
+		createPostMutation.mutate({ id: 30, title, content });
```

Comment:
The id is not needed in the object u want to send, because the server generates an id for the new post. Also for the update feature u will need unique ids, so this wont work, I think u should change it.